gips/0083.md

```diff
@@ -0,0 +1,240 @@
+
+`substreams tools` will contain a way to validate those attestations, and help with investigation.  This tool will take the content being disputed (BlockScopedData) from the replay.log, which contain the attestations over the content within, and validate their output, in order to compare against providers, and end up slashing an Indexer.
+
+```
+substreams tools validate-attestation --indexer-address 0x123123123 --attestation eth:120398102398102938109283012983 --replay-log-file ./path/to/replay.log
+```
+
+This command will process all messages in the provided replay log and validate their attestations. Replay logs can be generated using  `substreams gui --with-replay`.
+
+The Payment Gateway can keep an in-memory mapping will link the operator key to the staking key.
+
+- The `setOperator` function ([Staking.sol#L226](https://github.com/graphprotocol/contracts/blob/ce3ec16484dacc89a1b9cf08455256be830790e3/packages/contracts/contracts/staking/Staking.sol#L226)) links the operator key to the staking key and emits the `SetOperator` event ([IStakingBase.sol#L99](https://github.com/graphprotocol/contracts/blob/main/packages/contracts/contracts/staking/IStakingBase.sol#L99)).
+
+Such a system can hydrate from the Network Subgraph.
+
+## Arbitration
+
+Amendments to GIP-0009 (_Arbitration Charter_) are proposed, as part of this GIP, to include the necessary provisions for Substreams arbitration.
+
+Assignment of Arbiters is not in scope for this specification, and should be dealt with independently if the need arises.
+
+> INFORMATIVE: The Core Developers of Substreams committed to providing support for arbitration cases involving Substreams. Many third-party competent teams are also able and willing to help.
+
+
+### Tools
+
+THIS SECTION IS NON-NORMATIVE. Different tools can exist to do those operations. We are listing some for informational purposes here.
+
+For Indexers, in order to register a new Substreams provider on-chain, and show up on the Payment Gateway, you can use commands such as:
+
+```
+substreams network register --operator-wallet 0x123123123 --operator-priv-key-file my.key --service substreams-v1 --endpoint mainnet.eth.example.com --provider-logo https://www.example.com/static/mylogo.png --provider-id example-company --provider-name "Example Company" --network ethereum
+```
+
+This command allows providers to register their service, specifying their operator wallet, private key file, the service type (`substreams-v1`), the endpoint URL, logo, name, and the supported network. The registration data is signed by the operator's key.  An update to an existing registration can be achieved by re-running the command with updated parameters, based on the `service`, `providerId` and `network` tripled.  A corresponding `unregister` or `revoke` command would be added to allow providers to take down their service from the front-end.
+
+
+# Dependencies & Backwards Compatibility
```

Comment:
This might not be the best place, but generally think it would be really nice to add link/doc for the indexers on how to deploy and run Substreams alongside their existing subgraph services